bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -271,4 +276,32 @@ int kepler_irq_trace(struct trace_event_raw_softirq *ctx)
     return 0;
 }
 
+// count read page cache
+SEC("kprobe/mark_page_accessed")
+int kprobe__mark_page_accessed(struct pt_regs *ctx)
+{
+    u32 cur_pid = bpf_get_current_pid_tgid();
+    struct process_metrics_t *process_metrics;
+    process_metrics = bpf_map_lookup_elem(&processes, &cur_pid);
+    if (process_metrics)
+    {
+        process_metrics->page_cache_hit ++;
+    }
+    return 0;
+}
+
+// count write page cache
+SEC("kprobe/mark_buffer_dirty")
```

Comment:
`set_page_dirty`? this will work non buffered ops too

Reply:
I got the page cache hit example from the libbpf examples in [https://github.com/iovisor/bcc/blob/master/libbpf-tools/cachestat.c](https://github.com/iovisor/bcc/blob/master/libbpf-tools/cachestat.c).

But I agree that `set_page_dirty` might be better, since it is the function used to mark entire pages as dirty when data is modified.

Reply:
The results are consistent/(very similar) after changing to `set_page_dirty`.

```
--------
Comm:kepler, Total VM: 2347, TLB Flush: 2347, Page Cahe Hit: 0.002
Comm:stress-ng-vm-rw, Total VM: 4115, TLB Flush: 4115, Page Cahe Hit: 1728.728
Comm:stress-ng, Total VM: 135, TLB Flush: 135, Page Cahe Hit: 0.284
--------
Comm:kepler, Total VM: 2203, TLB Flush: 2203, Page Cahe Hit: 0.002
Comm:stress-ng-vm-rw, Total VM: 1858, TLB Flush: 1858, Page Cahe Hit: 6507.188
--------
Comm:kepler, Total VM: 7209, TLB Flush: 7209, Page Cahe Hit: 0.005
Comm:stress-ng-vm-rw, Total VM: 1791, TLB Flush: 1791, Page Cahe Hit: 6371.64
--------
Comm:kepler, Total VM: 2884, TLB Flush: 2884, Page Cahe Hit: 0.004
Comm:stress-ng-vm-rw, Total VM: 1700, TLB Flush: 1700, Page Cahe Hit: 6043.56
--------
Comm:kepler, Total VM: 3016, TLB Flush: 3016, Page Cahe Hit: 0.003
Comm:stress-ng-vm-rw, Total VM: 1578, TLB Flush: 1578, Page Cahe Hit: 5592.727
```

Reply:
I reverted back to `mark_buffer_dirty` because the integration test failed with `set_page_dirty` was not found.

https://github.com/sustainable-computing-io/kepler/actions/runs/6391448290/job/17347104002?pr=953

Reply:
can we try the same approach as before with sched_switch, i.e. first try `set_page_dirty`, if not there, then try `mark_buffer_dirty`

Reply:
Done! @rootfs can you review it again?

Reply:
coooooool!